### PR TITLE
fix-up 404 urls

### PR DESCRIPTION
**Pull request by Aresforchina**
fix some 404 urls
ref:#17649
@Rajakavitha1

### CODE REVIEW

**Comment by kbhawkey** [comment]
@Aresforchina . Thanks for this PR!
Going forward, it may be best to separate the blog changes from the other content updates.
Providing Page preview links is helpful.

https://deploy-preview-18006--kubernetes-io-master-staging.netlify.com/blog/2018/10/08/support-for-azure-vmss-cluster-autoscaler-and-user-assigned-identity/

https://deploy-preview-18006--kubernetes-io-master-staging.netlify.com/docs/tasks/federation/administer-federation/deployment/

File: content/en/docs/tasks/federation/administer-federation/deployment.md [hunk 1]
```diff
@@ -73,7 +73,7 @@ if you have 3 registered clusters and you create a Federated Deployment with
 `spec.replicas = 9`, then each Deployment in the 3 clusters will have
 `spec.replicas=3`.
 To modify the number of replicas in each cluster, you can specify
-[FederatedReplicaSetPreference](https://github.com/kubernetes/federation/blob/{{< param "githubbranch" >}}/apis/federation/types.go)
+[FederatedReplicaSetPreference](https://github.com/kubernetes-retired/federation/blob/master/apis/federation/types.go{{< param "githubbranch" >}})
```
Comment (kbhawkey) [inline content/en/docs/tasks/federation/administer-federation/deployment.md]: @Aresforchina I just tried this link in the preview. Can you recheck if this works? Thanks.

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md [hunk 2]
```diff
@@ -333,6 +333,6 @@ For the acs-engine (the unmanaged variety) on Azure docs can be found here: [[9]
 
 13) https://github.com/Azure/acs-engine/tree/master/examples/kubernetes-msi-userassigned
 
-14) https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/auth/azure_auth.go
+14) https://github.com/kubernetes/kubernetes/blob/feature-serverside-apply/pkg/cloudprovider/providers/azure/auth/azure_auth.go
```
Comment (kbhawkey) [inline content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: 👍

File: content/en/docs/tasks/federation/administer-federation/deployment.md (hunk 1 above)
Comment (Aresforchina) [reply content/en/docs/tasks/federation/administer-federation/deployment.md]: @kbhawkey Of course, I've already rechecked it

Comment (kbhawkey) [reply content/en/docs/tasks/federation/administer-federation/deployment.md]: Hello @Aresforchina . I retried the link again. This link worked for me, _not_ the link above:

https://github.com/kubernetes-retired/federation/blob/master/apis/federation/types.go

This code is retired. In the short term, the link should be fixed on the page.
I logged an issue (some time ago) to update the federation pages to the new federation information (https://github.com/kubernetes-sigs/kubefed).

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: @Aresforchina . I checked the links on the whole page. The links are set up a bit differently than expected. I found that these links did _not_ work from within the text:
- link number 3
- link number 14

Can you check in the text to make sure that the links match what is in the link Reference section? 😃 

The Kubernetes website local links forward correctly in the text, but also could be linked up in the Reference section as well.

Comment (Aresforchina) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: @kbhawkey  OK!I have alreay fix them .Thanks for your suggestion.

File: content/en/docs/tasks/federation/administer-federation/deployment.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/tasks/federation/administer-federation/deployment.md]: Did you remove the federation change from this PR?

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md [hunk 3]
```diff
@@ -264,7 +264,7 @@ With v1.12 we bring user assigned managed identity support for Kubernetes. With
...
+The acs-engine sets up the cluster with the required configuration. The /etc/kubernetes/azure.json file provides a way for the cluster components (eg: kube-apiserver) to gather configuration on how to access the cloud resources. In a user managed identity cluster there is a value filled with the key as `UserAssignedIdentityID`. This value is filled with the client id of the user assigned identity created by acs-engine or provided by the user, however the case may be. The code which does the authentication for Kubernetes on azure can be found here [[14]](https://github.com/kubernetes/kubernetes/tree/feature-serverside-apply/pkg/cloudprovider/providers/azure). This code uses Azure adal packages to get authenticated to access various resources in the cloud. In case of user assigned identity the following API call is made to get new token:
```
Comment (kbhawkey) [inline content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Is this link closer to the original?
https://github.com/kubernetes/kubernetes/blob/feature-serverside-apply/pkg/cloudprovider/providers/azure/auth/azure_auth.go

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md [hunk 4]
```diff
@@ -65,7 +65,7 @@ In addition to the above interfaces, the storage services from the cloud provide
 
 ## Azure cloud provider implementation and VMSS
 
-In the Azure cloud provider, for every type of cluster we implement, there is a VMType option which we specify. In case of VMSS, the VM type is “vmss”.  The provisioning software (acs-engine, in future AKS etc.) would setup these values in /etc/kubernetes/azure.json file. Based on this type, various implementations would get instantiated [[3]](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_vmss.go)
+In the Azure cloud provider, for every type of cluster we implement, there is a VMType option which we specify. In case of VMSS, the VM type is “vmss”.  The provisioning software (acs-engine, in future AKS etc.) would setup these values in /etc/kubernetes/azure.json file. Based on this type, various implementations would get instantiated [[3]](https://github.com/kubernetes/kubernetes/tree/feature-serverside-apply/pkg/cloudprovider/providers/azure)
```
Comment (kbhawkey) [inline content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Change to this link?
https://github.com/kubernetes/kubernetes/blob/feature-serverside-apply/pkg/cloudprovider/providers/azure/azure_vmss.go

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Check out line 314 and make sure it matches the link in line 68.

Comment (tengqm) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Folks, please try avoid pointers to the master branch which is under constant change. References to source code in general should be avoided. If we do have a real need to do so, please use pointer to a specific tag in the git repo.

Comment (kbhawkey) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Are you suggesting other changes? The links to `cloudprovider` seem susceptible to change.

Comment (tengqm) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: I was wondering something like although `feature-serverside-apply` might not be a good choice. It smells like a temporary working branch for a feature named "server-side apply". When that feature is merged, the branch may go away. I may be wrong if the `feature-serverside-apply` is the only branch to reference. If we have a choice, we should use tags like `1.16.0`.

Comment (kbhawkey) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Right. I am not familiar with what is happening in the cloud provider area.

Comment (Aresforchina) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: @ I did not find the cloud provider like azure in tags of 1.16.0.  should I look for for it in other tags?

Comment (tengqm) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Yup. That is exactly where the problem is. We may need more digging into the upstream code to understand the correct repo/tag to point to. @feiskyer can you help?

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md [hunk 5]
```diff
@@ -65,7 +65,7 @@ In addition to the above interfaces, the storage services from the cloud provide
 
 ## Azure cloud provider implementation and VMSS
 
-In the Azure cloud provider, for every type of cluster we implement, there is a VMType option which we specify. In case of VMSS, the VM type is “vmss”.  The provisioning software (acs-engine, in future AKS etc.) would setup these values in /etc/kubernetes/azure.json file. Based on this type, various implementations would get instantiated [[3]](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_vmss.go)
+In the Azure cloud provider, for every type of cluster we implement, there is a VMType option which we specify. In case of VMSS, the VM type is “vmss”.  The provisioning software (acs-engine, in future AKS etc.) would setup these values in /etc/kubernetes/azure.json file. Based on this type, various implementations would get instantiated [[3]](https://github.com/kubernetes/kubernetes/tree/feature-serverside-apply/pkg/cloudprovider/providers/azure/azure_vmss.go)
```
Comment (feiskyer) [inline content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: the newest link is https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md [hunk 6]
```diff
@@ -311,7 +311,7 @@ For the acs-engine (the unmanaged variety) on Azure docs can be found here: [[9]
 
 2) /docs/concepts/architecture/cloud-controller/
 
-3)  https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/azure/azure_vmss.go
+3)https://github.com/kubernetes/kubernetes/blob/feature-serverside-apply/pkg/cloudprovider/providers/azure/azure_vmss.go
```
Comment (feiskyer) [inline content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: https://github.com/kubernetes/kubernetes/blob/release-1.17/staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss.go

File: content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md (hunk 2 above)
Comment (feiskyer) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: it is https://github.com/kubernetes/kubernetes/tree/release-1.17/staging/src/k8s.io/legacy-cloud-providers/azure/auth

Comment (feiskyer) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: Thanks for fixing those links

Comment (tengqm) [reply content/en/blog/_posts/2018-10-08-support-for-azure-vmss.md]: @feiskyer Thanks for the help. That is a good URL (permanent and valid) to use.

**Comment by Aresforchina** [comment]
@feiskyer I  have modified these broken urls. Thanks for your help.

**Comment by kbhawkey** [comment]
Thanks!
/approve

**Pull request merged**